frontend/src/component/admin/apiToken/ApiTokenIcon/ApiTokenIcon.test.tsx

```diff
@@ -1,7 +1,62 @@
 import { render } from 'utils/testRenderer';
 import { screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import { ApiTokenIcon } from './ApiTokenIcon';
+import { ApiTokenIcon, isOrphanedToken } from './ApiTokenIcon';
+
+describe('isOrphanedToken', () => {
+    it('should be true for wildcard tokens with secret indicating it is orphaned', () => {
+        expect(
+            isOrphanedToken({
+                secret: '[]:development.be7536c3a160ff15e3a92da45de531dd54bc1ae15d8455c0476f086b',
+                projects: ['*'],
+                project: '*',
+            }),
+        ).toBe(true);
+        expect(
+            isOrphanedToken({
+                secret: 'test:development.be7536c3a160ff15e3a92da45de531dd54bc1ae15d8455c0476f086b',
+                projects: ['*'],
+                project: '*',
+            }),
+        ).toBe(true);
+    });
+
+    it('should be false for true wildcard tokens', () => {
+        expect(
+            isOrphanedToken({
+                secret: '*:development.be7536c3a160ff15e3a92da45de531dd54bc1ae15d8455c0476f086b',
+                projects: ['*'],
+                project: '*',
+            }),
+        ).toBe(false);
+    });
+
+    it('should be false for secrets in v1 format', () => {
+        expect(
+            isOrphanedToken({
+                secret: 'be44368985f7fb3237c584ef86f3d6bdada42ddbd63a019d26955178',
+                projects: ['*'],
+                project: '*',
+            }),
+        ).toBe(false);
+    });
+
+    it('should be false if there are projects assigned to a token', () => {
+        expect(
+            isOrphanedToken({
+                secret: 'test:development.be7536c3a160ff15e3a92da45de531dd54bc1ae15d8455c0476f086b',
+                projects: ['test'],
+                project: 'test',
+            }),
+        ).toBe(false);
+        expect(
+            isOrphanedToken({
+                secret: '[]:development.be7536c3a160ff15e3a92da45de531dd54bc1ae15d8455c0476f086b',
+                projects: ['a', 'b'],
+            }),
+        ).toBe(false);
+    });
```

Comment:
What if a token only has one project left but was made for multiple tokens?  Will this pass? (I think it will now, but just to be sure)

```suggestion
          expect(
              isOrphanedToken({
                  secret: '[]:development.be7536c3a160ff15e3a92da45de531dd54bc1ae15d8455c0476f086b',
                  projects: ['a'],
              }),
          ).toBe(false);
    });
```